src/common/grealpath.h

```diff
@@ -20,7 +23,35 @@ static inline gchar *g_realpath(const char *path)
 #define PATH_MAX 4096
 #endif
   char buffer[PATH_MAX] = { 0 };
-  if(realpath(path, buffer))
+  char* res = realpath(path, buffer);
+  
+  if ( errno == EACCES ) {
+    printf("Read or search permission was denied for a component of file_name: %s\n", path);
+    exit(EXIT_FAILURE);
```

Comment:
NOOOOO! Why would you call `exit`? If you really think that error handling is required then imitate what glib does and use GError to communicate that to the user.

Reply:
We're talking about functionality that is called on the initialization of darktable. This happens before the UI starts up. How can we react instead of prompting the reason to the terminal?
If I don't exit here the program will either die a few lines later because it tries to load .so files at invalid pathes or it will look weird because it does not find optional resources. Either way the error should be shown before the UI initializes.

Do you have a better proposal?